File: frontend/src/api/publicSector.ts
import { apiClient } from '@/api/client';

// ---- Public Owner Mode ----

export interface PublicOwnerModeData {
  id: string;
  organization_id: string;
  mode_type: string;
  is_active: boolean;
  governance_level: string;
  requires_committee_review: boolean;
  requires_review_pack: boolean;
  default_review_audience: string[] | null;
  notes: string | null;
  activated_at: string | null;
  created_at: string;
  updated_at: string;
}

// ---- Review Packs ----

export interface ReviewPackData {
  id: string;
  building_id: string;
  generated_by_user_id: string | null;
  pack_version: number;
  status: string;
  sections: Record<string, unknown> | unknown[] | null;
  content_hash: string | null;
  review_deadline: string | null;
  circulated_to: Array<{ org_name?: string; role?: string; sent_at?: string }> | null;
  notes: string | null;
  generated_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface ReviewPackCreatePayload {
  notes?: string | null;
  review_deadline?: string | null;
}

export interface CirculatePayload {
  recipients: Array<{ org_name: string; role: string; sent_at: string }>;
}

// ---- Committee Packs ----

export interface CommitteePackData {
  id: string;
  building_id: string;
  committee_name: string;
  committee_type: string;
  pack_version: number;
  status: string;
  sections: Record<string, unknown> | unknown[] | null;
  procurement_clauses: Array<{ clause: string; [key: string]: unknown }> | null;
  content_hash: string | null;
  decision_deadline: string | null;
  submitted_at: string | null;
  decided_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface CommitteePackCreatePayload {
  committee_name: string;
  committee_type: string;
  decision_deadline?: string | null;
  procurement_clauses?: Array<{ clause: string; [key: string]: unknown }> | null;
}

// ---- Decision Traces ----

export interface DecisionTraceData {
  id: string;
  pack_type: string;
  pack_id: string;
  reviewer_name: string;
  reviewer_role: string | null;
  reviewer_org_id: string | null;
  decision: string;
  conditions: string | null;
  notes: string | null;
  evidence_refs: Array<Record<string, unknown>> | null;
  confidence_level: string | null;
  decided_at: string;
  created_at: string;
}

export interface DecisionTraceCreatePayload {
  reviewer_name: string;
  reviewer_role?: string | null;
  reviewer_org_id?: string | null;
  decision: string;
  conditions?: string | null;
  notes?: string | null;
  evidence_refs?: Array<Record<string, unknown>> | null;
  confidence_level?: string | null;
  decided_at: string;
}

// ---- Governance Signals ----

export interface GovernanceSignalData {
  id: string;
  organization_id: string;
  building_id: string | null;
  signal_type: string;
  severity: string;
  title: string;
  description: string | null;
  source_entity_type: string | null;
  source_entity_id: string | null;
  resolved: boolean;
  resolved_at: string | null;
  created_at: string;
}

// ---- API ----

export const publicSectorApi = {
  // Public Owner Mode
  getPublicOwnerMode: async (orgId: string): Promise<PublicOwnerModeData> => {
    const response = await apiClient.get<PublicOwnerModeData>(`/organizations/${orgId}/public-owner-mode`);
    return response.data;
  },

  // Review Packs
  listReviewPacks: async (buildingId: string): Promise<ReviewPackData[]> => {
    const response = await apiClient.get<ReviewPackData[]>(`/buildings/${buildingId}/review-packs`);
    return response.data;
  },

  generateReviewPack: async (buildingId: string, payload: ReviewPackCreatePayload): Promise<ReviewPackData> => {
    const response = await apiClient.post<ReviewPackData>(`/buildings/${buildingId}/review-packs`, payload);
    return response.data;
  },

  circulateReviewPack: async (packId: string, payload: CirculatePayload): Promise<ReviewPackData> => {
    const response = await apiClient.post<ReviewPackData>(`/review-packs/${packId}/circulate`, payload);
    return response.data;
  },

  // Committee Packs
  listCommitteePacks: async (buildingId: string): Promise<CommitteePackData[]> => {
    const response = await apiClient.get<CommitteePackData[]>(`/buildings/${buildingId}/committee-packs`);
    return response.data;
  },

  generateCommitteePack: async (
    buildingId: string,
    payload: CommitteePackCreatePayload,
  ): Promise<CommitteePackData> => {
    const response = await apiClient.post<CommitteePackData>(`/buildings/${buildingId}/committee-packs`, payload);
    return response.data;
  },

  // Decision Traces
  listDecisionTraces: async (packType: string, packId: string): Promise<DecisionTraceData[]> => {
    const response = await apiClient.get<DecisionTraceData[]>(`/decision-traces/${packType}/${packId}`);
    return response.data;
  },

  recordDecision: async (packId: string, payload: DecisionTraceCreatePayload): Promise<DecisionTraceData> => {
    const response = await apiClient.post<DecisionTraceData>(`/committee-packs/${packId}/decide`, payload);
    return response.data;
  },

  // Governance Signals
  listGovernanceSignals: async (orgId: string, buildingId?: string): Promise<GovernanceSignalData[]> => {
    const response = await apiClient.get<GovernanceSignalData[]>(`/organizations/${orgId}/governance-signals`, {
      params: buildingId ? { building_id: buildingId } : undefined,
    });
    return response.data;
  },

  resolveGovernanceSignal: async (signalId: string): Promise<GovernanceSignalData> => {
    const response = await apiClient.post<GovernanceSignalData>(`/governance-signals/${signalId}/resolve`);
    return response.data;
  },
};

File: backend/app/schemas/public_sector.py
from __future__ import annotations

from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

# ---- PublicOwnerOperatingMode ----


class PublicOwnerModeCreate(BaseModel):
    mode_type: str  # municipal | cantonal | federal | public_foundation | mixed
    is_active: bool = True
    governance_level: str = "standard"  # standard | enhanced | strict
    requires_committee_review: bool = False
    requires_review_pack: bool = True
    default_review_audience: list[str] | None = None
    notes: str | None = None
    activated_at: datetime | None = None


class PublicOwnerModeRead(BaseModel):
    id: UUID
    organization_id: UUID
    mode_type: str
    is_active: bool
    governance_level: str
    requires_committee_review: bool
    requires_review_pack: bool
    default_review_audience: list[str] | None
    notes: str | None
    activated_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


# ---- MunicipalityReviewPack ----


class ReviewPackCreate(BaseModel):
    notes: str | None = None
    review_deadline: date | None = None


class ReviewPackRead(BaseModel):
    id: UUID
    building_id: UUID
    generated_by_user_id: UUID | None
    pack_version: int
    status: str
    sections: dict | list | None
    content_hash: str | None
    review_deadline: date | None
    circulated_to: list[dict] | None
    notes: str | None
    generated_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CirculateRequest(BaseModel):
    recipients: list[dict]  # [{org_name, role, sent_at}]


# ---- CommitteeDecisionPack ----


class CommitteePackCreate(BaseModel):
    committee_name: str
    committee_type: str  # municipal_council | building_committee | procurement_committee | technical_commission | other
    decision_deadline: date | None = None
    procurement_clauses: list[dict] | None = None


class CommitteePackRead(BaseModel):
    id: UUID
    building_id: UUID
    committee_name: str
    committee_type: str
    pack_version: int
    status: str
    sections: dict | list | None
    procurement_clauses: list[dict] | None
    content_hash: str | None
    decision_deadline: date | None
    submitted_at: datetime | None
    decided_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


# ---- ReviewDecisionTrace ----


class DecisionTraceCreate(BaseModel):
    pack_type: str | None = None  # committee | municipal_review — set by endpoint
    pack_id: UUID | None = None  # set by endpoint from path param
    reviewer_name: str
    reviewer_role: str | None = None
    reviewer_org_id: UUID | None = None
    decision: str  # approved | rejected | deferred | modified | abstained
    conditions: str | None = None
    notes: str | None = None
    evidence_refs: list[dict] | None = None
    confidence_level: str | None = None
    decided_at: datetime


class DecisionTraceRead(BaseModel):
    id: UUID
    pack_type: str
    pack_id: UUID
    reviewer_name: str
    reviewer_role: str | None
    reviewer_org_id: UUID | None
    decision: str
    conditions: str | None
    notes: str | None
    evidence_refs: list[dict] | None
    confidence_level: str | None
    decided_at: datetime
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


# ---- GovernanceSignal ----


class GovernanceSignalCreate(BaseModel):
    organization_id: UUID
    building_id: UUID | None = None
    signal_type: str
    severity: str = "info"
    title: str
    description: str | None = None
    source_entity_type: str | None = None
    source_entity_id: UUID | None = None


class GovernanceSignalRead(BaseModel):
    id: UUID
    organization_id: UUID
    building_id: UUID | None
    signal_type: str
    severity: str
    title: str
    description: str | None
    source_entity_type: str | None
    source_entity_id: UUID | None
    resolved: bool
    resolved_at: datetime | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
